File: Backend/ticket-booking/src/main/java/com/cinema/ticket_booking/dto/response/BookingResponse.java
package com.cinema.ticket_booking.dto.response;

import com.cinema.ticket_booking.enums.BookingStatus;
import lombok.Builder;
import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

import java.io.Serializable;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BookingResponse implements Serializable {
    private static final long serialVersionUID = 1L;

    private String id;
    private String bookingCode;
    private BookingStatus status;

    // Thông tin suất chiếu
    private String showtimeId;
    private String movieId;
    private String movieTitle;
    private String moviePosterUrl;
    private LocalDateTime startTime;
    private String cinemaName;
    private String cinemaAddress;
    private String screenName;
    private String screenType;

    // Ghế đã đặt
    private List<SeatItem> seats;

    // Combo
    private List<ComboItem> combos;

    // Thanh toán
    private BigDecimal subtotal; // Tổng vé + combo (sau khuyến mãi hệ thống, trước voucher)
    private BigDecimal totalOriginalAmount; // Tổng giá niêm yết ban đầu
    private BigDecimal promotionDiscountAmount; // Số tiền được giảm từ hệ thống
    private String appliedPromotionName; // Tên chương trình khuyến mãi hệ thống đã áp dụng
    private String voucherCode;
    private BigDecimal discountAmount; // Số tiền giảm từ voucher
    private BigDecimal totalAmount; // Tổng sau cùng
    private String warningMessage; // Thông báo cảnh báo (ví dụ Voucher hết hạn)

    // QR Code (chỉ có khi status = PAID)
    private String qrCode;

    private LocalDateTime createdAt;
    private LocalDateTime expiresAt; // Hết hạn thanh toán

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class SeatItem implements Serializable {
        private static final long serialVersionUID = 1L;
        private String showtimeSeatId;
        private Character rowLabel;
        private Integer colNumber;
        private String seatType;
        private BigDecimal price;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ComboItem implements Serializable {
        private static final long serialVersionUID = 1L;
        private String comboId;
        private String comboName;
        private Integer quantity;
        private BigDecimal unitPrice;
        private BigDecimal subtotal;
    }

    // Tóm tắt ngắn dùng cho danh sách lịch sử đặt vé
    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Summary implements Serializable {
        private static final long serialVersionUID = 1L;
        private String id;
        private String bookingCode;
        private String movieTitle;
        private String movieId;
        private String moviePosterUrl;
        private LocalDateTime startTime;
        private String cinemaName;
        private String screenName;
        private String screenType;
        private String seats;
        private BigDecimal totalAmount;
        private BookingStatus status;
        private LocalDateTime createdAt;
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/BookingResponse.kt
package com.cinema.ticket_booking.data.model.response

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class BookingResponse(
    @SerializedName("id")
    var id: String? = null,

    @SerializedName("bookingCode")
    var bookingCode: String? = null,

    @SerializedName("status")
    var status: String? = null,

    @SerializedName("movieTitle")
    var movieTitle: String? = null,

    @SerializedName("moviePosterUrl")
    var moviePosterUrl: String? = null,

    @SerializedName("startTime")
    var startTime: String? = null,

    @SerializedName("cinemaName")
    var cinemaName: String? = null,

    @SerializedName("cinemaAddress")
    var cinemaAddress: String? = null,

    @SerializedName("screenName")
    var screenName: String? = null,

    @SerializedName("screenType")
    var screenType: String? = null,

    @SerializedName("subtotal")
    var subtotal: Double? = null,

    @SerializedName("discountAmount")
    var discountAmount: Double? = null,

    @SerializedName("totalAmount")
    var totalAmount: Double? = null,

    @SerializedName("totalOriginalAmount")
    var totalOriginalAmount: Double? = null,

    @SerializedName("promotionDiscountAmount")
    var promotionDiscountAmount: Double? = null,

    @SerializedName("appliedPromotionName")
    var appliedPromotionName: String? = null,

    @SerializedName("warningMessage")
    var warningMessage: String? = null,

    @SerializedName("qrCode")
    var qrCode: String? = null,

    @SerializedName("expiresAt")
    var expiresAt: String? = null,

    @SerializedName("seats")
    var seats: List<SeatItem>? = null,

    @SerializedName("combos")
    var combos: List<ComboItem>? = null,

    @SerializedName("pointsUsed")
    var pointsUsed: Long? = null,

    @SerializedName("pointDiscount")
    var pointDiscount: Double? = null,

    @SerializedName("remainingAmount")
    var remainingAmount: Double? = null
) : Parcelable {

    @Parcelize
    data class SeatItem(
        @SerializedName("rowLabel")
        val rowLabel: String? = null,

        @SerializedName("colNumber")
        val colNumber: Int = 0,

        @SerializedName("seatType")
        val seatType: String? = null,

        @SerializedName("price")
        val price: Double = 0.0,

        @SerializedName("showtimeSeatId")
        var showtimeSeatId: String? = null
    ) : Parcelable

    @Parcelize
    data class ComboItem(
        @SerializedName("comboName")
        val comboName: String? = null,

        @SerializedName("quantity")
        val quantity: Int = 0,

        @SerializedName("unitPrice")
        val unitPrice: Double = 0.0,

        @SerializedName("subtotal")
        val subtotal: Double = 0.0,

        @SerializedName("comboId")
        var comboId: String? = null
    ) : Parcelable
}
